src/exr-wrap/exr-wrap.js

```diff
@@ -1,24 +1,3359 @@
+                      return fetch(wasmBinaryFile, {
+                          credentials: "same-origin"
+                      }).then((function(response) {
+                          if (!response["ok"]) {
+                              throw "failed to load wasm binary file at '" + wasmBinaryFile + "'"
+                          }
+                          return response["arrayBuffer"]()
+                      })).catch((function() {
+                          return getBinary()
+                      }))
+                  }
+                  return new Promise((function(resolve, reject) {
+                      resolve(getBinary())
+                  }))
+              }
+
+              function doNativeWasm(global, env, providedBuffer) {
+                  if (typeof WebAssembly !== "object") {
+                      err("no native wasm support detected");
+                      return false
+                  }
+                  if (!(Module["wasmMemory"] instanceof WebAssembly.Memory)) {
+                      err("no native wasm Memory in use");
+                      return false
+                  }
+                  env["memory"] = Module["wasmMemory"];
+                  info["global"] = {
+                      "NaN": NaN,
+                      "Infinity": Infinity
+                  };
+                  info["global.Math"] = Math;
+                  info["env"] = env;
+
+                  function receiveInstance(instance, module) {
+                      exports = instance.exports;
+                      if (exports.memory) mergeMemory(exports.memory);
+                      Module["asm"] = exports;
+                      Module["usingWasm"] = true;
+                      removeRunDependency("wasm-instantiate")
+                  }
+                  addRunDependency("wasm-instantiate");
+                  if (Module["instantiateWasm"]) {
+                      try {
+                          return Module["instantiateWasm"](info, receiveInstance)
+                      } catch (e) {
+                          err("Module.instantiateWasm callback failed with error: " + e);
+                          return false
+                      }
+                  }
+
+                  function receiveInstantiatedSource(output) {
+                      receiveInstance(output["instance"], output["module"])
+                  }
+
+                  function instantiateArrayBuffer(receiver) {
+                      getBinaryPromise().then((function(binary) {
+                          console.log(`binary: ${binary}`);
+                          return WebAssembly.instantiate(binary, info)
+                      })).then(receiver).catch((function(reason) {
+                          err("failed to asynchronously prepare wasm: " + reason);
+                          abort(reason)
+                      }))
+                  }
+                  
+                  if (typeof wasmBinaryFile === "function") {
```

Comment:
@cbreak-black  this is one of the few other pieces I added, just to handle when the wasmBinaryFile is a function loaded by webpack.